### PR TITLE
Fix chat previews not working

### PR DESCRIPTION
**Pull request by frej4189**
- Chat preview packet handler now uses the correct name for fields from protocol definition (`packet.query` is now `packet.queryId`)
- Pass original message to `_signedChat` instead of previewed message
- Create message signature using previewed message

### CODE REVIEW

File: src/client/chat.js [hunk 1]
```diff
@@ -158,7 +158,7 @@ module.exports = function (client, options) {
         message,
         timestamp: options.timestamp,
         salt: options.salt,
-        signature: client.profileKeys ? client.signMessage(message, options.timestamp, options.salt) : Buffer.alloc(0),
+        signature: client.profileKeys ? client.signMessage(options.preview || message, options.timestamp, options.salt) : Buffer.alloc(0),
```
Comment (extremeheat) [inline src/client/chat.js:161]: Is this correct? The signature check on the server will fail with this.

**Review by extremeheat — CHANGES_REQUESTED** [review]

File: src/client/chat.js (hunk 1 above)
Comment (frej4189) [reply src/client/chat.js:161]: Then the server signature check is incorrect aswell. Vanilla server accepts this and will kick with "Invalid or missing signature" if we use message instead

File: src/server/chat.js [hunk 2]
```diff
@@ -42,6 +42,8 @@ module.exports = function (client, server, options) {
   const raise = (translatableError) => client.end(translatableError, JSON.stringify({ translate: translatableError }))
   const pending = new Pending()
 
+  if (!options.generatePreview) options.generatePreview = message => message
```
Comment (extremeheat) [inline src/server/chat.js:45]: needs to be added to docs

**Comment by rom1504** [comment]
CI failing

**Pull request merged**